ClemBot.Bot/bot/cogs/translate_cog.py

```diff
@@ -1,272 +1,275 @@
-        log.info(f'Output Lang Code: {output_lang}')
-
-        params = {
-            'api-version': '3.0',
-            'to': output_lang
-        }
-
-        body = [{
-            'text': text
-        }]
-
-        headers = {
-            'Ocp-Apim-Subscription-Key': bot_secrets.secrets.azure_translate_key,
-            'Ocp-Apim-Subscription-Region': 'global',
-            'Content-type': 'application/json',
-            'X-ClientTraceId': TRACE_ID
-        }
-
-        async with aiohttp.ClientSession() as session:
-            async with await session.post(url=TRANSLATE_API_URL, params=params, headers=headers, json=body) as resp:
-                response = json.loads(await resp.text())
-
-        embed = discord.Embed(title='Translate', color=Colors.ClemsonOrange)
-        name = 'Translated to ' + LANGUAGE_SHORT_CODE_TO_NAME[response[0]['translations'][0]['to'].lower()]
-        embed.add_field(name=name, value=response[0]['translations'][0]['text'], inline=False)
-        embed.add_field(name='Confidence Level:', value=response[0]['detectedLanguage']['score'], inline=True)
-        embed.add_field(name='Detected Language:', value=LANGUAGE_SHORT_CODE_TO_NAME[response[0]['detectedLanguage']['language']], inline=True)
-        await ctx.send(embed=embed)
-        return
-
-
-def is_valid_lang_code(input: str):
-    return input.lower() in LANGUAGE_SHORT_CODE_TO_NAME or input.lower() in LANGUAGE_NAME_TO_SHORT_CODE
-
-
-async def get_lang_code(self, ctx, input: str):
-    if input.lower() in LANGUAGE_SHORT_CODE_TO_NAME:
-        return input.lower()
-    else:
-        try:
-            return LANGUAGE_NAME_TO_SHORT_CODE[input.lower()]
-        except KeyError:
-            pages = get_language_list(self)
-            await self.bot.messenger.publish(Events.on_set_pageable_text,
-                                             embed_name='Languages',
-                                             field_title='Given language \'' + input + '\' not valid. Here are the available languages:',
-                                             pages=pages,
-                                             author=ctx.author,
-                                             channel=ctx.channel)
-
-
-def get_language_list(self):
-    langs = [f'{name} ({short})' for name, short in LANGUAGE_NAME_TO_SHORT_CODE.items()]
-    return ['\n'.join(i) for i in chunk_list(self, langs, CHUNK_SIZE)]
-
-
-def chunk_list(self, lst, n):
-    for i in range(0, len(lst), n):
-        yield lst[i:i + n]
-
-
-def setup(bot):
-    bot.add_cog(TranslateCog(bot))
+
+import json
+import logging
+import uuid
+import aiohttp
+import discord
+import discord.ext.commands as commands
+from discord.ext.commands.errors import UserInputError
+import bot.bot_secrets as bot_secrets
+import bot.extensions as ext
+from bot.consts import Colors
+from bot.messaging.events import Events
+
+log = logging.getLogger(__name__)
+session = aiohttp.ClientSession()
+class TranslateCog(commands.Cog):

```

Comment:
The ClientSession should not be instantiated here, it should be created within the cog's `__init__` method and should be properly disposed of when the cog is unloaded.

https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientSession.close

https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=cog_unload#discord.ext.commands.Cog.cog_unload